### ISSUE TITLE
Add piece meal warning removal for test dir

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#4402]
As @cvvergara is tirelessly working to remove compiler warnings from our code base (and actually finding the odd bug) and now changed strategy to PR against my PRs (good idea!), we thought it might be smart to enable individual tests to be compiled with warnings as errors, similar to all other modules.

But before that is being worked on, I'd like check that it's ok with everyone. I assume so, still..:)

**Comment by kevinkreiser** [comment]
yep adding werror for tests is fine with me. it might be a bit annoying to do though right? we already have the tests spliced up by dependencies, we'd have to have a whitelist of which test targets should add werror to their compile definitions? id rather that than having yet another splitting of test targets but maybe there is an even easier way im not thinking of?

**Comment by nilsnolde** [comment]
Done since a while
